app/controllers/concerns/with_organization.rb

```diff
@@ -6,4 +6,8 @@ def set_organization!
   def visit_organization!
     current_user.visit!(Organization.current)
   end
-end
\ No newline at end of file
+
+  def implicit_central?
+    !(request.first_subdomain == 'central') && Organization.central?
```

Comment:
Also, `first_subdomain` answers `nil` if subdomain is not set. 

So this whole method should be replaced by 

`request.first_subdomain.blank?`  

It is simpler,and also more performant - no db queries.
. 
You could even add it to the `Request` class.